ui/src/mixins/lazy-loading-filter.js

```diff
@@ -0,0 +1,178 @@
+import { decamelize } from 'humps'
+
+export default {
+  props: {
+    entity: {
+      default: '',
+      type: String
+    },
+
+    lazyLoadingProps: {
+      default: () => ({}),
+      type: Object
+    },
+
+    name: {
+      default: '',
+      type: String
+    },
+
+    useLazyLoading: {
+      type: Boolean
+    }
+  },
+
+  data () {
+    return {
+      filteredOptions: [],
+      hasFetchError: false,
+      isLoading: false,
+      isScrolling: false,
+      pagination: {
+        page: 1,
+        lastPage: null
+      },
+      search: null
+    }
+  },
+
+  computed: {
+    lazyLoadingPropsWithDefaults () {
+      const {
+        url,
+        params,
+        decamelizeFieldName
+      } = this.lazyLoadingProps
+
+      const defaultParams = {
+        limit: 48
+      }
+
+      return {
+        url: url || '',
+        params: {
+          ...defaultParams,
+          ...params
+        },
+        decamelizeFieldName: decamelizeFieldName === undefined ? true : decamelizeFieldName
+      }
+    },
+
+    hasFilteredOptions () {
+      return !!this.filteredOptions.length
+    },
+
+    virtualScrollClassName () {
+      return `virtual-scroll-${this.name}`
+    }
+  },
+
+  methods: {
+    async filterOptionsByStore (value) {
+      this.filteredOptions = []
+      this.search = value
+      this.pagination = {
+        page: 1,
+        lastPage: null
+      }
+      this.filteredOptions = await this.fetchOptions()
+    },
+
+    async fetchOptions () {
+      try {
+        if (!this.entity) throw new Error(this.missingPropMessage('entity'))
+        if (!this.name) throw new Error(this.missingPropMessage('name'))
```

Comment:
Será que não compensa ter um método para validar isto? e adicionar um obj do que é required? se amanha outra prop for required vai precisar outro if, pensei em algo assim:

```js
function validateRequiredFetchOptionsProps () {
  const requiredProps = {
    entity: this.entity,
    name: this.name
  }

  for (const key of requiredProps) {
    if (!requiredProps[key]) return throw new Error(this.missingPropMessage(key))
  }
}
```

O que acha?

Reply:
Obs: não testei xD

Reply:
Por enquanto, vamos manter assim, pois será muito difícil ter mais do que duas propriedades obrigatórias.